src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -738,8 +742,7 @@ def restore_disks(cmd, client, resource_group_name, vault_name, container_name,
     trigger_restore_properties = _get_trigger_restore_properties(rp_name, vault_location, _storage_account_id,
                                                                  _source_resource_id, target_rg_id,
                                                                  use_original_storage_account, restore_disk_lun_list,
-                                                                 rehydration_duration, rehydration_priority,
-                                                                 rp_list[0].tier_type)
+                                                                 rehydration_duration, rehydration_priority, rp_list)
```

Comment:
instead for passing whole rp list here I'd suggest to revert back to the older implementation. pass the tier type value only. just make use of ternary operator. for ex - 
rp_list[0].properties.recovery_point_tier_details is None ? None : rp_list[0].tier_type

Reply:
Are you saying i should use this ternary operator inside the function call ? like _get_trigger_restore_properties(rp_name......., rp_list[0].properties.recovery_point_tier_details is None ? None : rp_list[0].tier_type)

Reply:
yes. let that argument be exclusively for the tier type info. passing the whole list may create confusion.

Reply:
pls check teams once